### PR TITLE
Add remote mfa support with local agent forwarding

### PR DESCRIPTION
**Pull request by Joerger**
This PR implements the `prompt-mfa-challenge@goteleport.com` extension described in https://github.com/gravitational/teleport/pull/18032.

When a remote agent connection is discovered with support for this extension, it will be used instead of local mfa prompt. This only affects functionality for `tsh ssh -o "ForwardAgent local"`.

Note: this feature is not currently used for remote mfa login, but this can be done as well.

based off of https://github.com/gravitational/teleport/pull/19421

Updates https://github.com/gravitational/teleport/issues/1648

### CODE REVIEW

**Comment by jakule** [comment]
@Joerger The linter is failing on this one. You need to run `make fix-imports` and update one deprecated import to make it pass.

**Review by jakule — APPROVED** [review]

File: lib/client/extended_keyring.go [hunk 1]
```diff
@@ -474,6 +491,98 @@ func callKeyExtension(agent agent.ExtendedAgent) (*profile.Profile, *ForwardedKe
 	return &profile, &forwardedKey, nil
 }
 
+// KeyExtensionResponse is a request object for the prompt-mfa-challenge@goteleport.com extension.
```
Comment (AntonAM) [inline lib/client/extended_keyring.go:494]:
```suggestion
// promptMFAChallengeRequest is a request object for the prompt-mfa-challenge@goteleport.com extension.
```

File: lib/client/extended_keyring.go [hunk 2]
```diff
@@ -474,6 +491,98 @@ func callKeyExtension(agent agent.ExtendedAgent) (*profile.Profile, *ForwardedKe
...
+type promptMFAChallengeRequest struct {
+	// ProxyAddr is the teleport proxy address that this challenge originated from.
+	ProxyAddr string
+	// MFAChallengeBlob is a json encoded *proto.MFAAuthenticateChallenge.
+	MFAChallengeBlob []byte
+	// MFAChallengeOptsBlob is a json encoded *proto.PromptMFAChallengeOpts.
+	MFAChallengeOptsBlob []byte
+}
+
+// KeyExtensionResponse is a response object for the prompt-mfa-challenge@goteleport.com extension.
```
Comment (AntonAM) [inline lib/client/extended_keyring.go:504]:
```suggestion
// promptMFAChallengeResponse is a response object for the prompt-mfa-challenge@goteleport.com extension.
```

File: lib/client/mfa.go [hunk 3]
```diff
@@ -110,19 +113,36 @@ func (tc *TeleportClient) PromptMFAChallenge(ctx context.Context, proxyAddr stri
...
+	// First, check for a forwarded ssh agent that supports remote mfa challenges.
+	socketPath := os.Getenv(teleport.SSHAuthSock)
+	conn, err := agentconn.Dial(socketPath)
+	if err == nil {
+		defer conn.Close()
+		agentclient := agent.NewClient(conn)
+		supportedExtension, err := callQueryExtension(agentclient)
+		if err == nil && supportedExtension[promptMFAChallengeAgentExtension] {
+			resp, err := callPromptMFAChallengeExtension(agentclient, proxyAddr, c, opts)
+			if err == nil {
```
Comment (AntonAM) [inline lib/client/mfa.go:132]: Should we ignore this error? Previous ones makes sense, but here we're in territory "agent is forwarded, extension is supported", so feels like this error is now significant, maybe at least log it.

File: lib/client/extended_keyring.go [hunk 4]
```diff
@@ -474,6 +491,98 @@ func callKeyExtension(agent agent.ExtendedAgent) (*profile.Profile, *ForwardedKe
...
+		if err := json.Unmarshal(req.MFAChallengeBlob, &challengeReq); err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		var challengeOpts PromptMFAChallengeOpts
+		if err := json.Unmarshal(req.MFAChallengeOptsBlob, &challengeOpts); err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		mfaResp, err := promptMFAStandalone(context.Background(), &challengeReq, req.ProxyAddr, &challengeOpts)
```
Comment (AntonAM) [inline lib/client/extended_keyring.go:529]: Maybe there's a way to add context into handler definition from the beginning? And there I think we could have better chances to use it for something useful, like proper context cancellation or tracing in the future.

File: lib/client/extended_keyring.go [hunk 5]
```diff
@@ -474,6 +491,98 @@ func callKeyExtension(agent agent.ExtendedAgent) (*profile.Profile, *ForwardedKe
...
+		}
+
+		return ssh.Marshal(promptMFAChallengeResponse{
+			MFAChallengeResponseBlob: buf.Bytes(),
+		}), nil
+	}
+}
+
+// callPromptMFAChallengeExtension calls the prompt-mfa-challenge@goteleport.com
+// issue the given mfa challenge to the agent.
```
Comment (AntonAM) [inline lib/client/extended_keyring.go:546]: Not sure if it's just me, or wording of the comment doesn't sound 100% clear. Bringing attention just in case.

**Review by AntonAM — APPROVED** [review]
Nice work! Some minor comments.

**Pull request closed without merging**